marketplace/client.go

```diff
@@ -2,117 +2,30 @@ package marketplace
 
 import (
 	"bytes"
+	"context"
+	"errors"
 	"io"
 	"mime/multipart"
 	"net/http"
 	"os"
 	"path"
 
+	"github.com/Khan/genqlient/graphql"
 	"github.com/lifeomic/phc-sdk-go/client"
-
-	"github.com/mitchellh/mapstructure"
 )
 
 //go:generate go run github.com/Khan/genqlient
 
-const GET_PUBLISHED_APP_TILE_MODULE = `
-  query GetPublishedModule($id: ID!, $version: String) {
-    myModule(moduleId: $id, version: $version) {
-      title
-      description
-	  version
-	  source {
-		... on AppTile {
-		  id
-		}
-	  }
-	  iconV2 {
-		url
-		fileName
-		fileExtension
-	  }
-    }
-  }
-`
-
-const CREATE_DRAFT_MODULE = `
- mutation CreateDraftModule($input: CreateDraftModuleInput!) {
-   createDraftModule(input: $input) {
-     id
-   }
- }
-`
-
-const SET_APP_TILE = `
-  mutation SetAppTile($input: SetPublicAppTileDraftModuleSourceInput!) {
-	setPublicAppTileDraftModuleSource(input: $input) {
-	  moduleId
-	}
-  }
-`
-
-const PUBLISH_MODULE = `
-  mutation PublishModule($input: PublishDraftModuleInputV2!) {
-	publishDraftModuleV2(input: $input) {
-	  id
-	  version {
-		version
-	  }
-	}
-  }
-`
-
-const START_IMAGE_UPLOAD = `
-  mutation StartImageUpload($input: StartUploadInput!) {
-	startUpload(input: $input) {
-	  id
-	  url
-	  fields
-	}
-  }
-`
-
-const FINALIZE_IMAGE_UPLOAD = `
-  mutation FinalizeImageUpload($input: FinalizeUploadInput!) {
-	finalizeUpload(input: $input) {
-	  moduleId
-	}
-  }
-`
-
 const GRAPHQL_URL = "marketplace-service:deployed/v1/marketplace/authenticated/graphql"
 
 type MarketplaceClient struct {
-	phcClient *client.Client
-}
-
-type appTileModule struct {
-	Title       string
-	Description string
-	Version     string
-	Source      struct {
-		Id string
-	}
-	IconV2 *struct {
-		Url           string
-		FileName      string
-		FileExtension string
-	}
+	phcClient *client.LambdaClient
+	gqlClient graphql.Client
 }
 
-func (marketplace *MarketplaceClient) getAppTileModule(id string) (*appTileModule, error) {
-	res, err := marketplace.phcClient.Gql(GRAPHQL_URL, GET_PUBLISHED_APP_TILE_MODULE, map[string]interface{}{"id": id})
-	if err != nil {
-		return nil, err
-	}
-	var data struct {
-		MyModule *appTileModule
-	}
-	err = mapstructure.Decode(res, &data)
-	if err != nil {
-		return nil, err
-	}
-	return data.MyModule, nil
+func (marketplace *MarketplaceClient) getAppTileModule(id string) (*AppTileModule, error) {
+	resp, err := GetPublishedModule(context.Background(), marketplace.gqlClient, id, "")
+	return &resp.MyModule.AppTileModule, err
```

Comment:
If there is an error, `resp` will be nil, and trying to reference `resp.MyModule.AppTitleModule` will cause a panic.
```suggestion
        if err != nil {
                return nil, err
        }
	return &resp.MyModule.AppTileModule, nil
```

Reply:
Fixed. Thanks!